<a href="https://colab.research.google.com/github/juan-lawrie/Repaso-Ciencias-de-datos/blob/main/Copia_de_01_%5BPr%C3%A1ctica%5D_Variables%2C_identificadores%2C_tipos_de_datos_y_operadores.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# 📌 Tema 1 Práctica: Variables, identificadores, tipos de datos y operadores


❗Antes de comenzar con la clase, hacete una copia de este documento así podés editarlo y guardar los cambios.

Hacé clic acá arriba, donde dice "Copiar en Drive".

Luego seguís trabajando en esa copia y te queda todo guardado en tu Drive.

---

## 📝 Ejercicio 1

Se desea registrar la información de una estudiante universitaria:

- Nombre completo: Paula García
- Edad: 22
- Ciudad de residencia: Santo Tomé
- Carrera: Ciencia de Datos
- Promedio: 8.5
- Materias aprobadas: 15
- Tiene beca: Sí
- Realizó prácticas: No

Guardá estos datos en variables usando los tipos de datos e identificadores adecuados:

In [ ]:
nombre_completo = 'Paula Garcia'
edad = 22
Ciudad_de_residencia = 'Santo Tomé'
Carrera = 'Ciencias de datos'
Promedio = 8.5
Tiene_Beca = True
Realizo_prácticas = False



## 📝 Ejercicio 2

A continuación se presentan varias situaciones cotidianas. Para cada una, respondé:

1. ¿Qué tipo de dato usarías para representar esta situación?
2. ¿Qué identificador utilizarías para representar este dato?

Situaciones:
1. El precio de una entrada para un concierto.
2. La cantidad de entradas vendidas en un evento.
3. La fecha en la que una persona realiza una compra en una tienda online.
4. El nombre de una película.
5. La calificación de una película en una escala de 1 a 10.
6. La temperatura máxima de un día.
7. Si una persona tiene activa la membresía en un gimnasio o no.

✏️ Respuestas:

1. Entero o flotante. precio_entrada_concierto
2. Entero. cantidad_entradas_vendidas
3. date. fecha_compra_realizada
4. string. nombre_pelicula
5. float. calificacion_pelicula
6. float. Temperatura_maxima_del_día
7. Booleano . membresía_activa_gimnasio

## 📝 Ejercicio 3

Observá el código en la celda a continuación:

In [ ]:
precio_producto1 = 1500
precio_producto2 = 2500
precio_producto3 = 3000

cantidad_producto1 = 3
cantidad_producto2 = 2
cantidad_producto3 = 5

ventas_producto1 = precio_producto1 * cantidad_producto1
ventas_producto2 = precio_producto2 * cantidad_producto2
ventas_producto3 = precio_producto3 * cantidad_producto3

total_ventas = ventas_producto1 + ventas_producto2 + ventas_producto3
media_ventas = total_ventas / 3

print(media_ventas)

8166.666666666667


📌 Preguntas:

1. ¿Cuántas variables se utilizan en el código? ¿Cuáles son y qué representa cada una?
2. ¿Qué tipos de datos se emplean?
3. Explicá con tus palabras lo que hace el código.

✏️ Respuestas:

1. Se utilizan 12 variables en el codigo. Son: precio_producto1 ,precio_producto2 ,precio_producto3,,cantidad_producto1 ,cantidad_producto2 ,cantidad_producto3,ventas_producto1 , ventas_producto2, ventas_producto3, total_ventas, media_ventas . precio_producto1,precio_producto2, precio_producto3 representan los precios de los productos.cantidad_producto1, cantidad_producto2, cantidad_producto3 representan la cantidad vendida de esos productos .ventas_producto1, ventas_producto2, ventas_producto3 representan la multiplicacion entre las primeras tres variables con sus correspondientes variables de cantidad. total_ventas representa la suma entre la suma de ventas_producto1, ventas_producto2 y ventas_producto3. media_ventas representa la divsion de total_ventas divido 3.

2. Tipos de datos enteros
3. En el codigo hay un precio definido para las variables de producto, con una cantidad determinada para cada producto que luego en cada variable de venta se multiplica el precio del producto con su respectiva cantidad. En total_venta se suma cada variable de ventas_producto osea(ventas_producto1, ventas_producto2 y ventas_producto3). Y luego se calcula la media con la variable de media_ventas que divide total_ventas dividido 3.

## 📝 Ejercicio 4

Un equipo de fútbol ofreció un descuento del 5% en las entradas para su próximo partido de local.

El precio original de cada entrada es de $5.000 y se vendieron 12.000 entradas.

Tareas:
- Calculá el monto de descuento que se aplica a cada entrada.
- Calculá el precio final de una entrada después de aplicar el descuento.
- Calculá la recaudación total obtenida por la venta de entradas con el descuento.

Completá el código para realizar los cálculos:

In [ ]:
# Guardá los datos en variables
precio_original = 5000
descuento = 5
entradas_vendidas = 12000

#precio_original * descuento / 100

# Calculá el descuento y el precio final
monto_descuento = 250
precio_final = 5000-250

# Calculá la recaudación total
recaudacion_total =precio_final * entradas_vendidas

# Mostrá el monto de descuento, el precio final y la recaudación total
print(monto_descuento)
print(precio_final)
print(recaudacion_total)

250
4750
57000000


Los costos totales del partido (apertura del estadio, seguridad, logística, etc.) ascienden a $40.500.000.

Sabiendo esto, calculá la ganancia neta (ingresos totales - costos totales) del evento:

In [ ]:
costos_totales = 40500000
recaudacion_total = 57000000
ganancia_neta = recaudacion_total - costos_totales

print(ganancia_neta)

16500000


Si el equipo espera aumentar la asistencia en un 7% para el próximo partido manteniendo el mismo precio con descuento:

- ¿Cuál sería la nueva recaudación total?
- ¿Cómo impactaría esto en la ganancia neta?

In [ ]:

nueva_asistencia = entradas_vendidas  * 7 / 100 + entradas_vendidas
nueva_recaudacion_total = precio_final * 12840
nueva_ganancia_neta = nueva_recaudacion_total - costos_totales

print(nueva_asistencia)
print(nueva_recaudacion_total)
print(nueva_ganancia_neta)

12840.0
60990000
20490000


## 📝 Ejercicio 5

A continuación, se presentan varias expresiones. Respondé las siguientes preguntas para cada una de ellas:

1. ¿En qué orden se realizan las operaciones?
2. Identificá los operadores y los operandos.

Expresiones:
1. `pi * radio ** 2`
2. `densidad * gravedad * profundidad`
4. `(G * masa_1 * masa_2) / (distancia ** 2)`
5. `(cateto_a ** 2 + cateto_b ** 2) ** 0.5`
6. `peso / (altura ** 2)`

✏️ Respuestas:

1. Primero la potencia y despues la multiplicacion. Operando: pi, radio y 2. Operador * y ** .

2. El orden de los factores no altera el producto, si calculas cualquiera de las dos multiplicaciones primero el resultado no cambia. Operando: densidad, gravedad, profundidad . Operador: * .

3. Se resuelvo primero lo que esta dentro de los parentesis y luego lo que esta afuera (la division en este caso) . Operando: G, masa_1, masa_2, distancia y 2. Operadores: *, ** y / .

4. Se realiza primero lo que esta dentro de los parentesis empezando por las potencias, luego la suma y luego lo que esta afuera de los mismos (en este caso hay una potencia afuera de los parentesis). Operando: cateto_a , 2, cateto_b, 2 y 0,5. Operadores: ** y +
5. Se realiza primero lo que esta dentro del parentesis y luego lo que esta afuera del mismo. Operando: peso, altura y 2. Operadores: ** y / .

## 📝 Ejercicio 6

Un servicio de streaming tiene tres tipos de suscripción con los siguientes datos mensuales:


| Tipo de Suscripción | Costo Mensual ($) | Número de Suscriptores |
|---------------------|------------------|------------------------|
| Básico             | 6.000            | 500.000                |
| Estándar           | 10.000           | 200.000                |
| Premium            | 13.500           | 100.000                |


Tareas:
- Calculá los ingresos totales por tipo de suscripción.
- Calculá el ingreso total del servicio de streaming.

Corregí todos los errores en el código para que los cálculos sean correctos:

In [ ]:
# Precios y suscriptores de cada plan
precio_basico = 6000
suscriptores_basico = 500000

precio_estandar = 10000
subcriptores_estandar = 200000

precio_premium = 13500
suscriptores_premium = 100000

In [ ]:
# Ingresos totales por plan


ingreso_basico = precio_basico * suscriptores_basico
ingreso_estandar = precio_estandar * subcriptores_estandar
ingreso_premium = precio_premium  * suscriptores_premium
ingreso_total = ingreso_basico + ingreso_estandar + ingreso_premium

print(ingreso_basico)
print(ingreso_estandar)
print(ingreso_premium)
#print(f"El ingreso total es {ingreso_total}")



3000000000
2000000000
1350000000
El ingreso total es 6350000000


In [ ]:
# Ingreso total del servicio
ingreso_total = ingreso_basico + ingreso_premium + ingreso_estandar

print(f"El ingreso total es {ingreso_total}")

El ingreso total es 6350000000


Si se ofrece un descuento del 10% en la suscripción Premium, ¿cuál será el nuevo ingreso de este tipo de suscripción y el total del servicio?

In [ ]:
precio_premium = 13500
descuento_precio_premium = precio_premium - precio_premium * 10 /100

suscriptores_premium = 100000 *  (precio_premium - precio_premium * 10 /100)

print(descuento_precio_premium)
print(suscriptores_premium)


12150.0
1215000000.0


## 📝 Ejercicio 7

En un proyecto, se registran los costos en tres fases: Planificación, Ejecución y Evaluación.

Los costos de cada fase son:

- Planificación: \$500.000
- Ejecución: \$1.500.000
- Evaluación: \$300.000

Tareas:
- Calculá el costo total del proyecto sumando los costos de cada fase.
- Calculá el porcentaje que representa cada fase respecto del costo total del proyecto.

Completá el código para realizar los cálculos:

In [ ]:
# Costos de cada fase
costo_planificacion = 500000
costo_ejecucion = 1500000
costo_evaluacion = 300000

In [ ]:
# Costo total del proyecto

costo_planificacion = 500000
costo_ejecucion = 1500000
costo_evaluacion = 300000
costo_total = costo_planificacion + costo_ejecucion + costo_evaluacion

print(costo_total)

2300000


In [ ]:
# Porcentaje de cada fase con respecto al costo total
porcentaje_planificacion = (costo_planificacion / costo_total) * 100
porcentaje_ejecucion = (costo_ejecucion / costo_total) * 100
porcentaje_evaluacion= (costo_evaluacion / costo_total) * 100

print(porcentaje_planificacion)
print(porcentaje_ejecucion)
print(porcentaje_evaluacion)


21.73913043478261
65.21739130434783
13.043478260869565


📌 Preguntas:
- ¿Cuánto debería dar la suma de los porcentajes de las tres fases?
- ¿Qué fase representa el mayor porcentaje del costo total y cuál es ese valor?

✏️ Respuestas:

1. La suma de los porcentajes de las tres fases deberia dar 100
2. La fase que representa el mayor porcentaje del costo total es la fase de ejecucion y es el %65.21739130434783

Si se asigna un presupuesto adicional de $200.000 para la fase de Evaluación, ¿cómo cambia su porcentaje respecto del total? El nuevo porcentaje respecto al total es del 20%

In [ ]:
costo_evaluacion = 500000
costo_planificacion = 500000
costo_ejecucion = 1500000
costo_total = costo_planificacion + costo_ejecucion + costo_evaluacion

porcentaje_evaluacion_nuevo= (costo_evaluacion  / costo_total  * 100)

print(porcentaje_evaluacion_nuevo)

20.0


## 📝 Ejercicio 8

Un equipo de natación está analizando los tiempos de sus tres nadadores en una competencia de 100 metros estilo libre.

Los tiempos (en segundos) registrados para los tres nadadores son:

- Nadador 1: 52.3 s
- Nadador 2: 54.1 s
- Nadador 3: 50.8 s

Tareas:
- Calculá la media de tiempo de los nadadores en la prueba de 100 metros.
- Calculá la diferencia entre el tiempo del nadador más lento y el más rápido.

In [ ]:
tiempo_nadador1 = 52.3
tiempo_nadador2 = 54.1
tiempo_nadador3 = 50.8

Media_nadadores =  (tiempo_nadador1 + tiempo_nadador2 + tiempo_nadador3) / 3

print(Media_nadadores )



52.4


Si el objetivo del equipo es que cada nadador baje su tiempo en un 5%, calculá los nuevos tiempos proyectados:

In [ ]:
tiempos_nadadores = [52.3 , 54.1 , 50.8]

tiempos_proyectado_nadador1 =  tiempos_nadadores[0] - (tiempos_nadadores[0] * 5) / 100
tiempos_proyectado_nadador2 = tiempos_nadadores[1] - (tiempos_nadadores[1] * 5) / 100
tiempos_proyectado_nadador3 = tiempos_nadadores[2] - (tiempos_nadadores[2] * 5) / 100

Media_nadadores =  (tiempos_proyectado_nadador1 + tiempos_proyectado_nadador2 + tiempos_proyectado_nadador3) /3
print(tiempos_proyectado_nadador1)
print(tiempos_proyectado_nadador2)
print(tiempos_proyectado_nadador3)
print(Media_nadadores)

49.684999999999995
51.395
48.26
49.78


Si el récord de la prueba es de 46.91 segundos, determiná cuánto necesita mejorar el nadador más rápido para alcanzarlo.

In [ ]:
tiempos_proyectado_nadador3 = tiempos_nadadores[2] - (tiempos_nadadores[2] * 5) / 100  - 46.91



print(f"El nadador mas rapido necesita mejorar: {tiempos_proyectado_nadador3} , para alcanzar al récord de la prueba que es 46.91 segundos")

El nadador mas rapido necesita mejorar: 1.3500000000000014 , para alcanzar al récord de la prueba que es 46.91 segundos


## 📝 Ejercicio 9

En una oficina se midió el consumo de energía (en kWh) de tres equipos durante cuatro semanas.

La siguiente tabla muestra el consumo registrado:

|Semana|Computadora (kWh)|Aire Acondicionado (kWh)|Cafetera (kWh)|Total Semana|  
|---|---|---|---|---|  
|**1**|7.2|36.3|24.0|...|  
|**2**|6.9|40.0|22.5|...|  
|**3**|7.5|38.5|23.9|...|  
|**4**|7.0|35.7|25.4|...|
|**Total Equipo**|...|...|...|

Tareas:
- Calculá el consumo total por equipo (total por columna) y por semana (total por fila).
- Completá la tabla con los totales que calculaste escribiendo donde están los 3 puntitos.

Completá el código para realizar los cálculos:

In [ ]:
# Computadora
computadora_semana1 = 7.2
computadora_semana2 = 6.9
computadora_semana3 = 7.5
computadora_semana4 = 7.0


# Aire acondicionado
aire_semana1 = 36.3
aire_semana2 = 40
aire_semana3 = 38.5
aire_semana4 = 35.7


# Cafetera
cafetera_semana1 = 24.0
cafetera_semana2 = 22.5
cafetera_semana3 = 23.9
cafetera_semana4 = 25.4


In [ ]:
# Consumo total por equipo
computadora_semana1 = 7.2
computadora_semana2 = 6.9
computadora_semana3 = 7.5
computadora_semana4 = 7.0

aire_semana1 = 36.3
aire_semana2 = 40
aire_semana3 = 38.5
aire_semana4 = 35.7

cafetera_semana1 = 24.0
cafetera_semana2 = 22.5
cafetera_semana3 = 23.9
cafetera_semana4 = 25.4


total_computadora = computadora_semana1 + computadora_semana2 + computadora_semana3 + computadora_semana4

total_aire = aire_semana1 + aire_semana2 + aire_semana3 + aire_semana4

cafetera_total = cafetera_semana1 + cafetera_semana2 + cafetera_semana3 + cafetera_semana4



print(total_computadora)
print(total_aire)
print(cafetera_total)


28.6
150.5
95.80000000000001


In [ ]:
# Consumo total por semana
total_semana1 = computadora_semana1 + aire_semana1 + cafetera_semana1
total_semana2 = computadora_semana2 + aire_semana2 + cafetera_semana2
total_semana3 = computadora_semana3 + aire_semana3 + cafetera_semana3
total_semana4 = computadora_semana4 + aire_semana4 + cafetera_semana4

print(total_semana1)
print(total_semana2)
print(total_semana3)
print(total_semana4)


67.5
69.4
69.9
68.1


## 📝 Ejercicio 10

Se ha lanzado una serie con tres temporadas. Los datos de audiencia y las calificaciones para estas tres temporadas son los siguientes:

|Temporada|Audiencia (millones)| Calificación (%)|
|--|--|--|
|1|2.51|91|
|2|3.80|96|
|3|4.96|96|

Tareas:
- Calculá el total de audiencia en las tres temporadas.
- Calculá la media de audiencia en las tres temporadas.
- Calculá la media de calificación en las tres temporadas.



In [ ]:
# Escribí tu código acá

Audiencia = [2510000, 3800000, 4960000]

Total_audiencia = Audiencia[0] + Audiencia[1] + Audiencia[2]

Media_audiencia = Total_audiencia / 3

Calificacion = [91,96,96]

Media_calificacion = (Calificacion[0] + Calificacion[1] + Calificacion[2]) /3

print(Total_audiencia)
print(Media_audiencia)
print(Media_calificacion)

11270000
3756666.6666666665
94.33333333333333


Este año se ha lanzado una cuarta temporada que tuvo una audiencia de 6.84 millones de personas y una calificación de 97%. Modificá tu código para incluir estos datos en los cálculos.

In [ ]:
Audiencia = [2510000, 3800000, 4960000, 6840000]

Total_audiencia = Audiencia[0] + Audiencia[1] + Audiencia[2] + Audiencia[3]

Media_audiencia = Total_audiencia / 3

Calificacion = [91,96,96,97]

Media_calificacion = (Calificacion[0] + Calificacion[1] + Calificacion[2] + Calificacion[3]) /4

print(Total_audiencia)
print(Media_audiencia)
print(Media_calificacion)

18110000
6036666.666666667
95.0


📌 Preguntas:
1. ¿Cómo cambia la media de audiencia al incluir la cuarta temporada?
2. ¿La media de calificación aumentó o disminuyó con la nueva temporada?
3. Si la audiencia continúa creciendo, ¿qué valores esperarías para una posible quinta temporada?

Para responder la pregunta 3 podés seguir estos pasos:
- Calculá el incremento de audiencia de una temporada a la siguiente.
- Determiná la media de esos incrementos.
- Sumá esta cantidad a la audiencia de la cuarta temporada.

In [ ]:
# Escribí tu código acá

Audiencia = [2510000, 3800000, 4960000, 6840000]

Incremento_audiencia_temporada1a2 = Audiencia[1] - Audiencia[0]
Incremento_audiencia_temporada2a3 =  Audiencia[2] - Audiencia[1]
Incremento_audiencia_temporada3a4 = Audiencia[3] - Audiencia[2]

Media_incremento_audiencia_temporada1a2 = Incremento_audiencia_temporada1a2 / 2

Media_incremento_audiencia_temporada2a3 = Incremento_audiencia_temporada2a3 / 2

Media_incremento_audiencia_temporada3a4 = Incremento_audiencia_temporada3a4 / 2



print(Incremento_audiencia_temporada1a2)
print(Incremento_audiencia_temporada2a3)
print(Incremento_audiencia_temporada3a4)

print(Media_incremento_audiencia_temporada1a2)
print(Media_incremento_audiencia_temporada2a3)
print(Media_incremento_audiencia_temporada3a4)

print(Media_incremento_audiencia_temporada1a2 + Media_incremento_audiencia_temporada2a3 + Media_incremento_audiencia_temporada3a4 + Audiencia[3])

1290000
1160000
1880000
645000.0
580000.0
940000.0
9005000.0


✏️ Respuestas:

1. Pasa de una media de 3756666.6666666665 a una de 6036666.666666667 .
2. La media de calificacion aumento a 95%.
3. Esperaria 9.005.000 para una posible quinta temporada